### ISSUE TITLE
How to avoid high memory usage without ProcessPoolExecutor?

### ISSUE THREAD

**Issue by putsi** [ross/requests-futures#90]
The example code results in very high memory usage when sending ~10k requests if every request gets huge response:

```
from concurrent.futures import as_completed
from requests_futures.sessions import FuturesSession
with FuturesSession() as session:
    futures = [session.get(URL_WITH_HUGE_RESPONSE) for i in range(10000)]
    for future in as_completed(futures):
        resp = future.result()
        print(resp.json()['url'])
```

The documentation suggests that `ProcessPoolExecutor` could be used to release memory back to OS. However ProcessPoolExecutor cannot be used when requests-futures is used inside a Celery task due to `daemonic processes are not allowed to have children`.

Are there any other ways to release memory to OS after a response has been handled? I've tried using `del resp` and `del future` in the for-loop but it didn't help.

**Comment by ross** [comment]
ProcessPool would use just as much memory, it'll just be spread across multiple processes, and would likely actually be a bigger problem since it'd need to transfer it across shared processes to get back to the main thread/other threads. 

If you search closed issues there's a bunch of previous discussions on the topic. https://github.com/ross/requests-futures/issues?q=is%3Aissue+stream+is%3Aclosed

TL;DR is that what you're trying to do might not fit into the use-case that the library is built for 
https://github.com/ross/requests-futures/issues/54#issuecomment-364810475. E.g. if you're downloading lots of big files. In that case you're likely better off building something purpose built that submits jobs and gets and writes out the responses in the job rather than trying to return them to the caller in a future.
